### ISSUE TITLE
ASIO can receive reply before write success callback

### ISSUE THREAD

**Issue by jadamcrain** [dnp3/opendnp3#73]
This requires rework to state machines. This should actually simplify things. We can pretend that the transmitting state doesn't exist, but not do a new task until transmission has finished.

**Comment by jadamcrain** [comment]
Fixed on 2.0.x - The master and outstation state machines now start the response timers just before starting transmission.
